cli/context/store/store.go

```diff
@@ -259,12 +271,113 @@ func Export(name string, s Reader) io.ReadCloser {
 	return reader
 }
 
+const (
+	maxAllowedFileSizeToImport int64 = 10 << 20
+)
+
+type fileData interface {
+	get() ([]byte, error)
+}
+
+type zipFileData struct {
+	zf *zip.File
+}
+
+func (zfd *zipFileData) get() ([]byte, error) {
+	src, err := zfd.zf.Open()
+	if err != nil {
+		return nil, err
+	}
+	defer src.Close()
+	data, err := LimitedReadAll(src, maxAllowedFileSizeToImport)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+type tarFileData struct {
+	tr io.Reader
+}
+
+func (tfd *tarFileData) get() ([]byte, error) {
+	data, err := LimitedReadAll(tfd.tr, maxAllowedFileSizeToImport)
+	if err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func newZipFileData(zf *zip.File) fileData {
+	var zfd = new(zipFileData)
+	zfd.zf = zf
+	return zfd
+}
+
+func newTarFileData(tr io.Reader) fileData {
+	var tfd = new(tarFileData)
+	tfd.tr = tr
+	return tfd
+}
+
+// ImportType enum to represent import type
+type ImportType int
+
+const (
+	// Cli will import Docker context from current env
+	Cli ImportType = iota
+	// Tar will import Docker context from a tar file
+	Tar
+	// Zip will import Docker context from a zip file
+	Zip
+)
+
 // Import imports an exported context into a store
-func Import(name string, s Writer, reader io.Reader) error {
+func Import(name string, s Writer, reader io.Reader, importType ImportType) error {
+	switch importType {
+	case Zip:
+		return importZip(name, s, reader)
+	default: // Tar, Cli
```

Comment:
Why have both Cli and Tar if they have the same behavior?

Reply:
To be honest with you, I've struggled to choose one over the other as just being the only type. Was worried that it might have cause more confusion. Would you recommend just getting rid of one of those import types, if yes, which one? Or should there be only `importType.Zip` and rest is the default value.

Reply:
With @thaJeztah's great direction on his PR, I've made the same updates here. We don't need the `imporType` enum anymore, it's way more cleaner.